bundles/auth/org.eclipse.smarthome.auth.oauth2client/src/main/java/org/eclipse/smarthome/auth/oauth2client/internal/cipher/SymmetricKeyCipher.java

```diff
@@ -0,0 +1,167 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.auth.oauth2client.internal.cipher;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.security.GeneralSecurityException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.util.Base64;
+import java.util.Dictionary;
+import java.util.Hashtable;
+
+import javax.crypto.Cipher;
+import javax.crypto.KeyGenerator;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.auth.client.oauth2.StorageCipher;
+import org.osgi.service.cm.Configuration;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is a symmetric key encryption service for encrypting the OAuth tokens.
+ *
+ * @author Gary Tse - Initial Contribution
+ *
+ */
+@Component
+public class SymmetricKeyCipher implements StorageCipher {
+
+    public static final String CIPHER_ID = SymmetricKeyCipher.class.getName();
+    private final Logger logger = LoggerFactory.getLogger(SymmetricKeyCipher.class);
+
+    private static final String PID = SymmetricKeyCipher.class.getName();
```

Comment:
Same here - in ESH, we use `org.eclipse.smarthome.<serviceId>` as a PID for configurable services throughout the codebase. But anyhow, it seems you are not using this constant at all...

Reply:
Removed useless constant.